fire/SFMainMod.F90

```diff
@@ -737,15 +746,20 @@ subroutine rate_of_spread ( currentSite, ROS_torch, passive_crown_FI, heat_per_a
 
           ! calculate heat release per unit area (HPA)(kJ/m2), Eq 2 Scott & Reinhardt 2001
           ! and residence time (min), Eq 3 Scott & Reinhardt 2001
-          time_r = 12.595 / currentPatch%fuel_sav 
+          time_r = 12.595_r8 / currentPatch%fuel_sav
           heat_per_area = ir * time_r          
 
           ! calculate torching index based on wind speed and crown fuels 
           ! ROS for crown torch initation (m/min), Eq 18 Scott & Reinhardt 2001
-          ! TODO slevis: Are we missing a -phi_s after the -1?
-          ROS_torch = 1._r8 / (54.683_r8 * wind_reduce) * &
-    ((60._r8 * passive_crown_FI * currentPatch%fuel_bulkd * eps * q_ig / &
-      (heat_per_area * ir * xi) - 1._r8) / (c * beta_ratio**-e))**(1._r8 / b)
+          ! temp1 requires max(0... to avoid raising a negative value to a
+          ! fractional temp2 power.
+          ! TODO slevis: omitting "- phi_s" after the -1 bc phi_s = 0 for now.
+          !              @jkshuman recommended naming it slope_factor.
+          temp1 = max(0._r8, (60._r8 * passive_crown_FI * &
+  currentPatch%fuel_bulkd * eps * q_ig / (heat_per_area * ir * xi) - 1._r8) / &
+  (c * beta_ratio**-e))
+          temp2 = 1._r8 / b
+          ROS_torch = temp1**temp2 / (54.683_r8 * wind_reduce)
```

Comment:
I added temp1 and temp2 to make Eq 18 easier to read. Eq 18 took me a while to debug. What eluded me until this morning was that `temp2` could be less than 1 at the same time as `temp1 < 0`, which causes a floating point exception. I added the max(0... to temp1 and the model now runs!

I completed 12 months using this test with `fates_fire_active_crown_fire = 0` in the params file:
`SMS_Lm12_D_Mmpi-serial.1x1_brazil.I2000Clm50FatesCruRsGs.izumi_intel.clm-FatesFireLightningPopDens`
Next I'm rerunning with `fates_fire_active_crown_fire = 1` in the params file. I will update this post when I know.

Reply:
The test with `fates_fire_active_crown_fire = 1` also completed 12 months. As soon I will get a chance I will check whether running with vs without active crown fire gave different answers and will update this post.

Reply:
For now this test gives bit-for-bit the same answers with and without active crown fire.

Reply:
This post can be ignored for now because ROS_torch is not used.